packages/runtime/container-runtime/src/summary/documentSchema.ts

```diff
@@ -490,6 +472,34 @@ export class DocumentsSchemaController {
 			},
 		};
 
+		// Schema coming from document (summary). If it's a new container, make sure that it's
+		// equal to desired / session schema (it's one and the same in such case), as that's what will
+		// go into summary.
+		this.documentSchema = !existing
```

Comment:
It might be worth updating the comment above the `documentSchema` declaration. Right now it says "schema coming from document metadata (snapshot we loaded from)" but it should say something like "schema coming from document metadata (snapshot we loaded from), or if no document exists (this is a new document) then this is the same as desiredSchema".